line_string.go

```diff
@@ -3,6 +3,13 @@ package geom
 // LineString is a basic line type which is made up of two or more points that don't interect.
 type LineString [][2]float64
 
-func (ls LineString) SubPoints() [][2]float64 {
-	return ls
+// Points returns a slice of XY values
+func (ls *LineString) Points() [][2]float64 {
+	return *ls
```

Comment:
Not needed?  Why add the Pointer. For functions that don't modify the value, it's better to not pass a pointer. Not to mention that a slice is just a struct with a pointer to a backing array and a cap, and nil. Also, if you are passing a pointer, you need to check for nil. This code can panic with a nil pointer exception.